discord_connector.py

```diff
@@ -158,128 +162,183 @@ def _handle_test_connectivity(self, param):
+
+        channel = await guild.fetch_channel(channel_id)
+        self.save_progress("channel: {}".format(str(channel)))
+
+        message = await channel.fetch_message(message_id)
+        self.save_progress("message: {}".format(str(message)))
+
+        await self._client.close()
+        return message
+
+    def create_artifacts(self, message):
+
+        container_id = BaseConnector.get_container_id(self)
+        attachments = []
+        embeds = []
+
+        # do we need it there or rather in consts???
+        artifact = {
+            "container_id": container_id,
+            "name": "name",
+            "cef": {
+                "URL": "",
+                "type": "",
+                "Description": ""
+            }
+        }
+
+        self.save_progress("working on embeds")
+        for embed in message.embeds:
+            self.save_progress("embed: {}".format(embed.to_dict))
+            embeds.append(self.create_embed_artifact(embed, artifact))
+
+        self.save_progress("working on attachments")
+        for attachment in message.attachments:
+            self.save_progress("attachment: {}".format(attachment.to_dict()))
+            attachments.append(self.create_attachment_artifact(attachment, artifact))
+
+        return attachments, embeds
+
+    # convert to use strategy pattern???
+
+    def create_embed_artifact(self, embed, artifact):
+        artifact["name"] = f"embed: {embed.title}"
+        artifact["cef"]["URL"] = embed.url
+        artifact["cef"]["Description"] = embed.description
+        status, creation_message, artifact_id = BaseConnector.save_artifact(self, artifact)
+        return artifact_id
+
+    def create_attachment_artifact(self, attachment, artifact):
+        artifact["name"] = f"attachment: {attachment.filename}"
+        artifact["cef"]["URL"] = attachment.url
+        artifact["cef"]["Description"] = attachment.description
+        artifact["cef"]["Type"] = attachment.content_type
+        status, creation_message, artifact_id = BaseConnector.save_artifact(self, artifact)
+        return artifact_id
+
+    def parse_message(self, message, attachments, embeds):
+
+        # not sure if those ifs are necessary or if those are not breaking return types
+        return {
+            "message origin": {
+                "channel id": message.channel.id,
+                "channel name": message.channel.name,
+            },
+            "message data": {
+                "created at": str(message.created_at),
+                "edited at": str(message.edited_at) if message.edited_at is not None else "message was not edited",
+            },
+            "author data": {
+                "author id": message.author.id,
+                "author name": message.author.name,
+            },
+            "jump url": message.jump_url,
+            "flags": self.parse_message_flags(message) if [] else "no flags",
```

Comment:
`if []`? this will always be False and go to "no flags".

Reply:
It has been resolved in PAPP-34496 | resolving issues | parsing message is now up and running using filter